Classes/Global State Explorers/FLEXFileBrowserTableViewController.m

```diff
@@ -265,4 +300,82 @@ - (void)openFileController:(NSString *)fullPath
     self.documentController = controller;
 }
 
+- (void)fileBrowserRename:(UITableViewCell *)sender
+{
+    NSString *fullPath = nil;
+
+    NSIndexPath *indexPath = [self.tableView indexPathForCell:sender];
+    if (indexPath) {
+        NSString *subpath = [self.childPaths objectAtIndex:indexPath.row];
+        fullPath = [self.path stringByAppendingPathComponent:subpath];
+    } else {
+        indexPath = [self.searchDisplayController.searchResultsTableView indexPathForCell:sender];
+        fullPath = [self.searchPaths objectAtIndex:indexPath.row];
+    }
+
+    self.fileOperationController = [[FLEXFileBrowserFileRenameOperationController alloc] initWithPath:fullPath];
+    self.fileOperationController.delegate = self;
+    [self.fileOperationController show];
+}
+
+- (void)fileBrowserDelete:(UITableViewCell *)sender
+{
+    NSString *fullPath = nil;
+
+    NSIndexPath *indexPath = [self.tableView indexPathForCell:sender];
+    if (indexPath) {
+        NSString *subpath = [self.childPaths objectAtIndex:indexPath.row];
+        fullPath = [self.path stringByAppendingPathComponent:subpath];
+    } else {
+        indexPath = [self.searchDisplayController.searchResultsTableView indexPathForCell:sender];
+        fullPath = [self.searchPaths objectAtIndex:indexPath.row];
+    }
+
+    self.fileOperationController = [[FLEXFileBrowserFileDeleteOperationController alloc] initWithPath:fullPath];
+    self.fileOperationController.delegate = self;
+    [self.fileOperationController show];
+}
+
+- (void)reloadDisplayedPaths {
+    if (self.searchDisplayController.isActive) {
+        [self reloadSearchPaths];
+        [self.searchDisplayController.searchResultsTableView reloadData];
+    } else {
+        [self reloadChildPaths];
+        [self.tableView reloadData];
+    }
+}
+
+- (void)reloadChildPaths {
+    self.childPaths = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:self.path error:NULL];
+}
+
+- (void)reloadSearchPaths {
+    self.searchPaths = nil;
+    self.searchPathsSize = nil;
+
+    //clear pre search request and start a new one
+    [self.operationQueue cancelAllOperations];
+    FLEXFileBrowserSearchOperation *newOperation = [[FLEXFileBrowserSearchOperation alloc] initWithPath:self.path searchString:self.searchString];
+    newOperation.delegate = self;
+    [self.operationQueue addOperation:newOperation];
+}
+
+@end
+
+
+@implementation FLEXFileBrowserTableViewCell
```

Comment:
I will love that to work, but it doesn't. My theory is that Apple only wants us to use that for copy, cut, and paste, and they don't invoke that method unless is one of those three (maybe others, but not the custom one).

I will give it a try, but I don't think I will make it work without the custom cell and playing with the responser chain (the last part is “optional”, but I didn’t want to give the cell the responsibility of deleting/renaming its file).